src/Extensions/Diagnostics.HealthChecks/Internal/HttpHealthCheckParameters.cs

```diff
@@ -55,6 +58,10 @@ internal class HttpHealthCheckParameters : HealthCheckParameters
 					? scheme
 					: throw new ArgumentException("Invalid uri scheme", nameof(scheme));
 
+			Headers = headers == null
```

Comment:
Could you please make field type `IReadOnlyDicrionary` and then use a single static instance of the empty dictionary here to avoid allocations for each creation.